gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -300,6 +359,32 @@ def self.run_all_sig_blocks
     end
   end
 
+  # the module adder is adding the methods from the module getting_added to itself. we need to check that for all
+  # instance methods M on getting_added, M is not defined on any of adder's ancestors.
+  def self._included_extended_impl(adder, adder_ancestors, getting_added)
+    if !module_with_final?(adder) && !module_with_final?(getting_added)
+      return
+    end
+    add_module_with_final(adder)
+    install_hooks(adder)
+    _check_final_ancestors(adder, adder_ancestors - getting_added.ancestors, getting_added.instance_methods)
+  end
+
+  def self.enable_final_checks_for_include_extend
+    if @enabled_final_checks_for_include_extend
+      return
+    end
+    @enabled_final_checks_for_include_extend = true
+    old_included = T::Private::ClassUtils.replace_method(Module, :included) do |arg|
+      old_included.bind(self).call(arg)
+      ::T::Private::Methods._included_extended_impl(arg, arg.ancestors, self)
+    end
+    old_extended = T::Private::ClassUtils.replace_method(Module, :extended) do |arg|
+      old_extended.bind(self).call(arg)
+      ::T::Private::Methods._included_extended_impl(arg, arg.singleton_class.ancestors, self)
+    end
+  end
```

Comment:
Given notes below about `before` / `after` blocks and test idempotency, what do you think about having a way to undo this?

Reply:
Hm. That would require us to store the `old_{included, extended}` somewhere and then call `replace`. I think I'd rather just turn it on once for the tests and leave it on. I think we should wait for the (in my opinion unlikely) user ask for a way to turn this off once turned on.

Reply:
After reconsideration, I added this.